presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/caching/TransactionScopeCachingJdbcClient.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc.caching;
+
+import io.prestosql.plugin.jdbc.ForwardingJdbcClient;
+import io.prestosql.plugin.jdbc.JdbcClient;
+import io.prestosql.plugin.jdbc.JdbcColumnHandle;
+import io.prestosql.plugin.jdbc.JdbcIdentity;
+import io.prestosql.plugin.jdbc.JdbcTableHandle;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.util.Objects.requireNonNull;
+
+public class TransactionScopeCachingJdbcClient
+        extends ForwardingJdbcClient
+{
+    private final ConcurrentHashMap<JdbcTableHandle, List<JdbcColumnHandle>> getColumnsCache = new ConcurrentHashMap<>();
+
+    private final JdbcClient delegate;
+
+    public TransactionScopeCachingJdbcClient(JdbcClient delegate)
+    {
+        this.delegate = requireNonNull(delegate, "delegate is null");
+    }
+
+    @Override
+    protected JdbcClient getDelegate()
+    {
+        return delegate;
+    }
+
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        getColumnsCache.computeIfAbsent(tableHandle, jdbcTableHandle -> super.getColumns(session, jdbcTableHandle));
```

Comment:
`computeIfAbsent` returns the value, so we can return it directly instead of looking it up